subworkflows/local/sex_determination.nf

```diff
@@ -0,0 +1,47 @@
+//
+// Run sex determine
+//
+
+include { SAMTOOLS_DEPTH } from '../../modules/nf-core/samtools/depth/main'
+include { SEXDETERRMINE } from '../../modules/nf-core/sexdeterrmine/main'
+
+workflow RUN_SEXDETERMINE {
+
+    take:
+    // samtoolsdepth_input // channel: [ val(meta1), [ bam ], [ meta2 ] [ intervals_bed ] ]
+    sexdeterrmine_bam // channel: [ val(meta1), [ bam ] ]
+    sexdeterrmine_bed // channel: [ [ meta2 ] [ intervals_bed ] ]
+    sexdeterrmine_list // channel: [ val(meta), [ sample_list_file ] ]
+
+    main:
+    ch_versions       = Channel.empty()
+    ch_multiqc_files  = Channel.empty()
+
+
+    // Generate sex_determine input
+    // Break here the input not in the workflow
+
+    ch_samtoolsdepth_input = sexdeterrmine_bam
+                            .combine(sexdeterrmine_bed)
+                            .multiMap{
+                            meta, bam, meta2, bed ->
+                            bam       : [ meta, bam ]
+                            intervals : [ meta2, bed ]
+                                    }
+
```

Comment:
With multi reference mapping, this approach will no longer work. The BED file should be read in from the REFERENCE_INDEXING SWF, and will come with its own meta. You will need to combine the bams with the correct BED for the reference, before splitting. Have a look how we do this in other SWFs.